### PR TITLE
Adds temperature, top_k decoding, and top_p decoding to decode_seq2seq.py

### PR DESCRIPTION
**Pull request by Phirefly9**
This PR adds the capability to provide different token decoding strategies.

Pretty much a transplant of some code from https://github.com/huggingface/transformers/blob/master/examples/run_generation.py

### CODE REVIEW

File: src/biunilm/decode_seq2seq.py [hunk 1]
```diff
@@ -91,6 +91,12 @@ def main():
                         help="Batch size for decoding.")
     parser.add_argument('--beam_size', type=int, default=1,
                         help="Beam size for searching")
+    parser.add_argument('--temperature', type=float, default=1,
```
Comment (Phirefly9) [inline src/biunilm/decode_seq2seq.py]: remembered last night this needs to default to 0 to maintain the same behavior as the old code when disabled, I will update the PR today when I get the chance

**Comment by Phirefly9** [comment]
This is ready for review.

arguments were set so that results of the paper will not change by default when running eval.

File: src/pytorch_pretrained_bert/modeling.py [hunk 2]
```diff
@@ -1449,10 +1453,15 @@ def forward(self, input_ids, token_type_ids, position_ids, attention_mask, task_
             last_hidden = new_encoded_layers[-1][:, -1:, :]
             prediction_scores, _ = self.cls(
                 last_hidden, None, task_idx=task_idx)
+            prediction_scores = prediction_scores[:, -1, :] / (self.temperature if self.temperature > 0 else 1.)
```
Comment (donglixp) [inline src/pytorch_pretrained_bert/modeling.py]: `prediction_scores[:, -1, :]` reduces the dimension from 3 to 2, while other places assume the dim=3 (such as `prediction_scores[:, :, token_id].fill_(-10000.0)`).

Comment (Phirefly9) [reply src/pytorch_pretrained_bert/modeling.py]: good catch, I checked and prediction_scores only needed to be 3 dimensional for the not_predict_set if conditional, and is not used after that, so I modified the access to work properly with 2 dimensions.

unfortunately due to to way torch.multinomial works I cannot leave it as 3 dimensions, so had to make the modification that way
